client/me/concierge/reschedule/calendar-step.js

```diff
@@ -0,0 +1,76 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getConciergeSignupForm } from 'state/selectors';
+import { getCurrentUserLocale } from 'state/current-user/selectors';
+import { rescheduleConciergeAppointment } from 'state/concierge/actions';
+import AvailableTimePicker from '../shared/available-time-picker';
+import {
+	CONCIERGE_STATUS_BOOKING,
+	CONCIERGE_STATUS_BOOKED,
+	WPCOM_CONCIERGE_SCHEDULE_ID,
+} from '../constants';
+
+class CalendarStep extends Component {
+	static propTypes = {
+		appointmentId: PropTypes.string.isRequired,
+		availableTimes: PropTypes.array.isRequired,
+		currentUserLocale: PropTypes.string.isRequired,
+		onComplete: PropTypes.func.isRequired,
+		site: PropTypes.object.isRequired,
+		signupForm: PropTypes.object.isRequired,
+	};
+
+	onSubmit = timestamp => {
+		const { appointmentId } = this.props;
+
+		this.props.rescheduleConciergeAppointment(
+			WPCOM_CONCIERGE_SCHEDULE_ID,
+			appointmentId,
+			timestamp
+		);
+	};
+
+	componentWillUpdate( nextProps ) {
+		if ( nextProps.signupForm.status === CONCIERGE_STATUS_BOOKED ) {
+			// go to confirmation page if booking was successful
+			this.props.onComplete();
+		}
+	}
+
+	render() {
+		const { availableTimes, currentUserLocale, signupForm, site, translate } = this.props;
+
+		return (
+			<AvailableTimePicker
+				actionText={ translate( 'Reschedule to this date' ) }
+				availableTimes={ availableTimes }
+				currentUserLocale={ currentUserLocale }
+				disabled={ signupForm.status === CONCIERGE_STATUS_BOOKING }
+				description={ translate( 'Please select a day to reschedule your Concierge session.' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Please select a day to have your Concierge session.' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).